mesh/nurbs.cpp

```diff
@@ -3305,6 +3305,47 @@ void NURBSExtension::Set3DSolutionVector(Vector &coords, int vdim)
    }
 }
 
+void NURBSExtension::SetPatchToElements()
+{
+   const int np = GetNP();
+   patch_to_el.resize(np);
+
+   for (int e=0; e<el_to_patch.Size(); ++e)
+   {
+      patch_to_el[el_to_patch[e]].Append(e);
+   }
+}
+
+void NURBSExtension::SetPatchToBdrElements()
+{
+   const int nbp = GetNBP();
+   patch_to_bel.resize(nbp);
+
+   for (int e=0; e<bel_to_patch.Size(); ++e)
+   {
+      patch_to_bel[bel_to_patch[e]].Append(e);
+   }
+}
+
+const Array<int>& NURBSExtension::GetPatchElements(int patch)
+{
+   if (patch_to_el.size() == 0)
```

Comment:
This if-statement might slow down the code if this function is called often. I think a more efficient location to create `patch_to_el` might be after creating `el_to_patch` in `Generate2DElementDofTable()` for instance.  (As it inverts `el_to_patch` right?). Then as debug check a `MFEM_Assert` can be added here. @dylan-copeland have you looked into this?

Reply:
I agree that this is a good suggestion, although I doubt that the if-statement may affect performance. Checking the size of an array should be faster than using the returned array, but in any case it makes sense to create `patch_to_el` only after creating `el_to_patch`. This change has been committed.